File: Consecutive Numbers Sum/Consecutive_Numbers_Sum.py
# Runtime: 220 ms, faster than 49.53% of Python3 online submissions for Consecutive Numbers Sum.
# Memory Usage: 13.3 MB, less than 5.00% of Python3 online submissions for Consecutive Numbers Sum.

class Solution:
    def consecutiveNumbersSum(self, N: int) -> int:
        res = 0
        m = 1
        while True:
            temp = 2 * N - (m - 1) * m
            if temp <= 0:
                break
            if temp % (2 * m) is 0:
                res += 1
            m += 1
        return res

File: Consecutive Numbers Sum/Consecutive_Numbers_Sum.cpp
// 基本思路就是等差数据的求和 需要循环的次数可以计算一下 明显是sqrt量级的

// Runtime: 4 ms, faster than 100.00% of C++ online submissions for Consecutive Numbers Sum.
// Memory Usage: 8.1 MB, less than 100.00% of C++ online submissions for Consecutive Numbers Sum.

class Solution 
{
public:
    int consecutiveNumbersSum(int N) 
    {
        int res = 0;
        for (int m = 1; ; m++)
        {
            int temp = 2 * N - (m - 1) * m;
            
            if (temp <= 0)
                break;
            
            if (temp % (2 * m) == 0)
                res++;
        }
        return res;
    }
};